tests/test_attribute.py

```diff
@@ -53,13 +53,41 @@ async def test_simple_handler_rw(mocker: MockerFixture):
     update_display_mock = mocker.patch.object(attr, "update_display_without_process")
     set_mock = mocker.patch.object(attr, "set")
 
-    await attr.sender.put(mocker.ANY, attr, 1)
+    await attr.sender.put(attr, 1)
 
     update_display_mock.assert_called_once_with(1)
     # The Sender of the attribute should just set the value on the attribute
     set_mock.assert_awaited_once_with(1)
 
 
+class SimpleUpdater(Updater):
+    pass
+
+
+@pytest.mark.asyncio
+async def test_handler_initialise(mocker: MockerFixture):
+    handler = SimpleHandler()
+    handler_mock = mocker.patch.object(handler, "initialise")
+    attr = AttrR(Int(), handler=handler)
+
+    await attr.initialise(mocker.ANY)
+
+    # The handler initialise method should be called from the attribute
+    handler_mock.assert_called_once_with(mocker.ANY)
```

Comment:
This should make a named `MagicMock` to pass to `attr.initialise` and then check the `handler_mock` is called with that.
For assertions where you only care about some of the arguments you can use mocks to check those and fill the rest with `mocker.ANY`.